src/pip/_internal/commands/index.py

```diff
@@ -35,6 +35,21 @@ def add_options(self) -> None:
         self.cmd_opts.add_option(cmdoptions.pre())
         self.cmd_opts.add_option(cmdoptions.no_binary())
         self.cmd_opts.add_option(cmdoptions.only_binary())
+        self.cmd_opts.add_option(
+            "--latest",
+            dest="latest",
+            action="store_true",
+            default=False,
+            help="Print only the latest version of the given package.",
+        )
```

Comment:
If we drop this on for now I think this can go in. But the show only latest logic would need much more discussion.